yeagerai/genlayer-simulator
# 236 - SIM-Add signatures to transactions

Fixes: 
- #236 
- #369 
- #353 
- #246 
- #371 
- #286 

# What
- Added transaction encoding, signing utils, and tests to the frontend.
- Added transaction signature to the frontend when calling a contract deployment and when calling a contract method
- Added new RPC endpoint `send_raw_transaction` that verifies the transaction signature and decodes the data 
- Refactored the integration tests to use the `send_raw_transaction` as well.

# Why

Our RPC server should be ethereum-compatible. One key feature towards that is to send transactions in the same way any account does it in ethereum.

The 

# Testing done



- run unit testing in frontend
- verify transaction signatures in the front end and backend

# Decisions made

- We keep a single lib in frontend. In this case, Viem does not add another
- I avoided to refactor the whole code fully, more improvements should be applied to the frontend to improve the way we are structuring the code.

# Checks

- [x] I have created an issue for this PR
- [x] I have set the PR name to the issue name

# User facing release notes
- The RPC server is not ethereum-compatible to send transactions and to deploy contracts. The method send_raw_transaction has been implemented to send serialized signed transactions. The previous way of sending transactions and deploying contracts has been deprecated and those methods are going to be removed in the next versions.

# Other notes
- More unit tests should be added both in the backend and in the frontend
